synapse/storage/databases/main/events_bg_updates.py

```diff
@@ -706,3 +710,187 @@ def get_rejected_events(
             )
 
         return len(results)
+
+    async def _chain_cover_index(self, progress: dict, batch_size: int) -> int:
+        """A background updates that iterates over all rooms and generates the
+        chain cover index for them.
+        """
+
+        current_room_id = progress.get("current_room_id", "")
+
+        # Have we finished processing the current room.
+        finished = progress.get("finished", True)
+
+        # Where we've processed up to in the room, defaults to the start of the
+        # room.
+        last_depth = progress.get("last_depth", -1)
+        last_stream = progress.get("last_stream", -1)
+
+        # Have we set the `has_auth_chain_index` for the room yet.
+        has_set_room_has_chain_index = progress.get(
+            "has_set_room_has_chain_index", False
+        )
+
+        if finished:
+            # If we've finished with the previous room (or its our first
+            # iteration) we move on to the next room.
+
+            def _get_next_room(txn: Cursor) -> Optional[str]:
+                sql = """
+                    SELECT room_id FROM rooms
+                    WHERE room_id > ?
+                        AND (
+                            NOT has_auth_chain_index
+                            OR has_auth_chain_index IS NULL
+                        )
+                    ORDER BY room_id
```

Comment:
This should essentially be random -- is there any (potential) benefit to trying to prioritize these "better"? I'm guessing no, but we have little idea how long this will take?

Reply:
Unfortunately I couldn't think of a stable ordering that was better. We could do big rooms first, but e.g. the number of state events changes over time. The alternative here would be to create a temporary table with all rooms, sort via some method, and then iterate over that. I'm not sure its worth it.

> I'm guessing no, but we have little idea how long this will take?

Not really, no. It was going at <0.1Hz (of state events) on jki.re, but that was before I did some optimisations so hopefully it'll go a bit  faster now.